Contact/view/frontend/layout/signup_index_index.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" layout="1column"
+    xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+    <body>
+        <referenceContainer name="content">
+            <block class="Gorilla\Contact\Block\Contact" name="signup" template="Gorilla_Contact::signup.phtml"/>
```

Comment:
There is no need to use a custom block class here. Instead, a View Model should be used.